### DISCUSSION TITLE
Internal Group ID from Messages

### DISCUSSION THREAD

**Discussion by mightym** [AsamK/signal-cli#656] [Q&A] [answered]
Sorry, I know this is not a bug. But I've been digging around the whole day how to figure this out with no success:

I'm getting via the DBUS service on `onMessageReceived` for the groupID a list of ints (bytes array). Looks like this: 
`[173, 151, 71, 53, 32, 62, 235, 210, 249, 242, 73, 225, 244, 161, 158, 193, 105, 253, 66, 4, 117, 167, 236, 100, 36, 232, 163, 194, 234, 25, 47, 162]`
And I wonder how I can convert it to the internal id string that looks like `rZdHNSA+69L58knk9HGewWn9QgR1p+xkJOijwuoZL6I=` I tried several ways to decode/encode that bytes array in python but never achieved that string. (thats the only group I'm having and I double checked it via `signal-cli -u XXXXXXXXXXXX listGroups  -d`)

Maybe someone has a hint for me. 

Thanks in advance and sorry for using the issues for my question :-/

**Comment by exquo** [comment, marked as answer]
This should do it:

```python
import base64
ba = [173, 151, 71, 53, 32, 62, 235, 210, 249, 242, 73, 225, 244, 161, 158, 193, 105, 253, 66, 4, 117, 167, 236, 100, 36, 232, 163, 194, 234, 25, 47, 162]
b = bytes(ba)
gid = base64.b64encode(b).decode()
print(gid)
```

See also #272.

**Reply by mightym** [reply]
Oh thanks a lot @exquo you made my day! :)

**Comment by exquo** [comment]
Out of curiosity, when are you getting this message on dbus? I've started `dbus-monitor`, sent a message to a group, and it has been received by members. But I don't see either `onMessageReceived` nor the integer byte array anywhere.

**Reply by mightym** [reply]
I was basically using this example here https://github.com/AsamK/signal-cli/wiki/DBus-service
And especially the python script from there. I can share you my docker file or give you access to my repo if you want? Feel free to get in touch.

**Reply by exquo** [reply]
Oh I see. The message is probably coming from `pydbus` then. Thanks!

**Reply by mightym** [reply]
Yes. And btw. one small thing on the example is outdated:
you have to get the DBUS via via the domain AND the path of the dbus interface:
So instead of:
`signal = bus.get('org.asamk.Signal')`
its
`signal = bus.get("org.asamk.Signal", "/org/asamk/Signal/_YOUR_PHONE_NUMBER")`

**Reply by exquo** [reply]
Thanks! Added to the wiki
